### ISSUE TITLE
[dash] Support autogenerated toc using jekyll-toc

### ISSUE THREAD

**Issue by chalin** [flutter/website#1259]
- [x] Setup use of `jekyll-toc`.
  - [x] Base setup and use of `jekyll-toc` - #1260
  - [x] Create path to ensure `no_toc` is supported - https://github.com/toshimaru/jekyll-toc/pull/50.
  - [x] Once the path is merged, revert to using gem (not from git).
- [x] [in progress] Strip out uses of manually inserted Kramdown TOCs: `* TOC {:toc}`. Related: #1337. Closed by #1261, #1341.
- [x] Ensure that pages that shouldn't have a TOC don't (by setting `toc: false` in the frontmatter) - #1261, ...
- [x] Ensure all pages with a TOC have:
  - Appropriate heading levels: e.g., there should be no \<h1>s, see #2126
  - [x] Headings (e.g., inside `card`s or \<aside>s) are marked `no_toc`.
- Ensure that link targets bring the page in view adjusted in a way that compensates for the main nav. See #1488
- [x] Setup use of a "scroll spy" to highlight TOC heading entry as section is scrolled to - #1271
- [x] Setup **Page Top** button in side TOC - #1271
- [x] Ensure that TOC entry links work (that the targets lead to the appropriate anchors) - #1264

Also see followup issue: TOC issues #1272.

cc @fertolg @Sfshaza @filiph

**Comment by fertolg** [comment]
I'm removing myself from this ticket since I don't think this applies to me, but please let me know if I'm wrong.

**Comment by chalin** [comment]
All done!
